src/modules/EnsonoBuild/exported/Publish-GitHubRelease.ps1

```diff
@@ -87,11 +92,18 @@ function Publish-GitHubRelease() {
 
     $fileError = $false
 
-    # if the artifactsList is empty, get all the files in the specified artifactsDir
-    # otherwise find the files that have been specified
+    # If the artifactsList is empty, get all the files in the specified
+    # artifactsDir otherwise find the files that have been specified
     if ($artifactsList.Count -eq 0) {
-        $artifactsList = Get-ChildItem -Path $artifactsDir -Recurse -File
+        if ($uploadArtifacts) {
+            $artifactsList = Get-ChildItem -Path $artifactsDir -Recurse -File
+        }
     } else {
+        if (!$uploadArtifacts) {
+            Write-Error "'uploadArtifacts' is set to false, but 'artifactsList' isn't empty..."
```

Comment:
maybe this is better as a warning...

Reply:
Yeah, I'll think on it, but I feel like if the user is supplying a list and they don't want to upload something has gone wrong